tests/unit/test_sampler.py

```
@@ -0,0 +1,209 @@
+    
+    @pytest.mark.asyncio
+    async def test_generate_location(self, data_paths):
+        """Test location generation."""
+        # Generate a location
+        location = await generate_location(
+            state=None,  # Any state
+            city=None,   # Any city
+            weighted=True,
+            make_api_call=False,
+            locations_path=data_paths["locations_path"],
+            json_path=data_paths["json_path"]
```

Ruff (0.12.2)

Use of `assert` detected

(S101)